packages/flutter/lib/src/widgets/animated_child_switcher.dart

```diff
@@ -62,8 +109,45 @@ class AnimatedChildSwitcher extends StatefulWidget {
   /// The duration over which to perform the cross fade using [FadeTransition].
   final Duration duration;
 
+  /// The builder function used to generate custom transitions for
+  /// [AnimatedChildSwitcher].
+  final AnimatedChildSwitcherTransitionBuilder transitionBuilder;
+
+  /// The builder function used to generate custom child layouts for
+  /// [AnimatedChildSwitcher].
+  final AnimatedChildSwitcherLayoutBuilder layoutBuilder;
+
   @override
   _AnimatedChildSwitcherState createState() => new _AnimatedChildSwitcherState();
+
+  /// The default transition algorithm used by [AnimatedChildSwitcher].
+  ///
+  /// The new child is given a [FadeTransition] which increases opacity as
+  /// the animation goes from 0.0 to 1.0, and decreases when the animation is
+  /// reversed.
+  ///
+  /// This is the default value for [transitionBuilder]. It implements
+  /// [AnimatedChildSwitcherTransitionBuilder].
+  static Widget defaultTransitionBuilder(Widget child, Animation<double> animation) {
+    return new FadeTransition(
+      opacity: animation,
+      child: child,
+    );
+  }
+
+  /// The default layout algorithm used by [AnimatedChildSwitcher].
+  ///
+  /// The new child is placed in a stack that sizes itself to match the largest
+  /// child or previous child. The children are centered on each other.
+  ///
+  /// This is the default value for [layoutBuilder]. It implements
+  /// [AnimatedChildSwitcherLayoutBuilder].
+  static Widget defaultLayoutBuilder(List<AnimatedChildSwitcherLayoutData> children) {
+    return new Stack(
+      children: children.map<Widget>((AnimatedChildSwitcherLayoutData child) => child.widget).toList(),
+      alignment: Alignment.center,
+    );
+  }
 }
 
 class _AnimatedChildSwitcherState extends State<AnimatedChildSwitcher> with TickerProviderStateMixin {
```

Comment:
addEntry's logic is a bit wrong for the case of _currentChild.widget == null. We should probably consider moving to a world where if widget.child is null, we set _currentChild to null rather than setting its widget to null and still creating a (useless) animation controller.

Reply:
Fixed.